### ISSUE TITLE
No HostAliases provided?

### ISSUE THREAD

**Issue by liuchintao** [tektoncd/pipeline#1750]
We have a private harbor registry in our k8s cluster which can not be accessed from Internet.

Now I want to push built image to harbor using domain name <docker.xxx.com>,
but container in pod can not resolve domain name from CoreDNS.
I think may be I need a feature in `PodTemplate` like [`HostAliases`](https://kubernetes.io/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases/)

# Expected Behavior

Target image was pushed successfully.

# Actual Behavior

DNS server can not resolve domain name.

lookup `docker.xxx.com` on `<dns svc endpoint>` : no such host

# Steps to Reproduce the Problem

These are my yamls.

secrets/harbor-secret.yaml

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: harbor-secret
  namespace: tektontest
  annotations:
    tekton.dev/docker-0: docker.xxx.com
type: kubernetes.io/basic-auth
stringData:
  username: <username>
  password: <password>
```

resource/img-resource.yaml

```yaml
apiVersion: tekton.dev/v1alpha1
kind: PipelineResource
metadata:
  name: img-pipeline-resource
  namespace: tektontest
spec:
  type: image
  params:
    - name: url
      value: docker.xxx.com/library/helloworld:0.1
```

git-img-admin.yaml

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: git-img-tekton
  namespace: tektontest
secrets:
  - name: git-usr-pwd-secret
  - name: harbor-secret
```

build-task.yaml

```yaml
apiVersion: tekton.dev/v1alpha1
kind: Task
metadata:
  name: build-gostudy-task
  namespace: tektontest
spec:
  inputs:
    resources:
      - name: go-gitlab
        type: git
    params:
      - name: pathToDockerfile
        description: The path to the dockerfile to build
        default: /workspace/go-gitlab/Dockerfile
      - name: pathToImgCtx
        description: The build context used by kaniko
        default: /workspace/go-gitlab
  outputs:
    resources:
      - name: buildImage
        type: image
  steps:
    - name: build-go
      image: golang:1.12.0
      workingDir: /workspace/go-gitlab
      command:
        - sh
      args: ["-c", "go build -o work"]
    - name: build-go-image
      image: kaniko-executor
      args:
        - --dockerfile=$(inputs.params.pathToDockerfile)
        - --context=$(inputs.params.pathToImgCtx)
        - --destination=$(outputs.resources.buildImage.url)
        - --insecure
```

**Comment by liuchintao** [comment]
@vdemeester Hi, maybe pipeline should support HostAliases or any other hosts related params.

Our k8s works on internal network, and we have private docker registry, 
but pipeline can not resolve registry's domain name.

I have tried many ways to work arround it, but they all failed.
It is a little possible to add registry's domain name into internal DNS server

**Comment by vdemeester** [comment]
@liuchintao indeed, I think it's fine to add `HostAliases` to the `PodTemplate`. I don't see any reason against it :wink:

**Comment by liuchintao** [comment]
@vdemeester Thanks for your reply, that's so cool 👍

**Comment by gejunqiang** [comment]
any update about this issue? I met the same problem recently.

**Comment by skk2142** [comment]
I also need this feature. I am trying to run a browser test and it requires a url other than localhost. `hostAliases` field allows for that:
```yaml
podTemplate:
    hostAliases:
    - ip: "127.0.0.1"
      hostnames:
        - "tekton.dev"
```
Please add this!

**Comment by sbwsg** [comment]
Given continued interest I'm reopening the issue.

@liuchintao @gejunqiang @skk2142 I don't think anyone's working on this right now but contributions are welcome.

**Comment by tlawrie** [comment]
I'd like to add my +1 for needing this as well. I feel the PodTemplate is the right place for this.

**Comment by xclud** [comment]
@sbwsg Please check out this PR.

https://github.com/tektoncd/pipeline/pull/3889

**Comment by sbwsg** [comment]
/close

Thanks a lot for tackling this @xclud!  This feature will be released as part of 0.24.
